### PR TITLE
SW-5184 Updated withdrawal reassignments to allow selecting zones

### PR DESCRIPTION
**Pull request by tommylau523**
Updated planting zones to be updated when a withdrawal is reassigned. 

Changes:

Show planting zones in To: Subzone because subzones under different planting zones can have the same name.

New reassignment screen. New subzones are not populated until the zone is selected. 

When the same zone is selected, it hides the same subzone option. 

When a subzone is selected, the zone is locked out, until subzone is cleared. This is because you can update the existing zone without updating subzones, and lead to confusion (especailly with name collisions).

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2504** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2504?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/NurseryRouter/ReassignmentRenderer.tsx [hunk 1]
```diff
@@ -27,75 +27,117 @@ const useStyles = makeStyles((theme: Theme) => ({
...
       }
-      updateReassignment('quantity', quantityValue);
+      setReassignment({ ...reassignment, quantity: quantityValue });
     };
 
+    if (column.key === 'newZone') {
+      const value = (
+        <Autocomplete
+          id={`newZone_${plantingId}`}
+          selected={selectedZone ?? undefined}
```
Comment (nickgraz) [inline src/scenes/NurseryRouter/ReassignmentRenderer.tsx:106]: I think this is already `undefined` if it is falsey

File: src/scenes/NurseryRouter/ReassignmentRenderer.tsx [hunk 2]
```diff
@@ -27,75 +27,117 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+          onChange={(newZoneValue: any) => {
+            if (newZoneValue.value) {
+              setReassignment({ ...reassignment, newZoneId: newZoneValue.value, newSubzoneId: undefined });
+            }
+          }}
+          isEqual={(option: any, selected: any) => option?.value === selected?.value}
```
Comment (nickgraz) [inline src/scenes/NurseryRouter/ReassignmentRenderer.tsx:112]: Can you replace these `any` usage with the ones used in the component props?

**Review by nickgraz — APPROVED** [review]
Looks good to me, a couple small nitpicks

**Comment by tommylau523** [comment]
### Merge activity

* **Apr 11, 6:44 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2504).

**Pull request merged**